pkg/sql/pg_catalog.go

```diff
@@ -1008,6 +1009,20 @@ func (r oneAtATimeSchemaResolver) getTableByID(id sqlbase.ID) (*TableDescriptor,
 	return table.Desc.TableDesc(), nil
 }
 
+func (r oneAtATimeSchemaResolver) getSchemaByID(
+	id sqlbase.ID,
+) (*sqlbase.ImmutableSchemaDescriptor, error) {
+	// TODO (rohany): This should use the descs.Collection.
```

Comment:
what happens if we don't at this point in time? we don't have caching? is there correctness issues with doing it this way atm?

Reply:
Nothing bad goes wrong, we'll just be doing a kv op every time. Work is in progress right now to add the appropriate API's to the descs.Collection.

Reply:
And so uncommitted schemas and the like show up?
(or should it not show up)

Reply:
Yeah, they'll show up (which is expected during a transaction)